include/net/net_pkt.h

```diff
@@ -124,7 +124,7 @@ struct net_pkt {
 #endif /* CONFIG_NET_IPV6_FRAGMENT */
 #endif /* CONFIG_NET_IPV6 */
 
-#if defined(CONFIG_NET_L2_IEEE802154) || defined(CONFIG_IEEE802154_RAW_MODE)
+#if defined(CONFIG_IEEE802154) || defined(CONFIG_IEEE802154_RAW_MODE)
```

Comment:
no CONFIG_IEEE802154 is not "more generic". CONFIG_IEEE802154 means one or more 15.4 drivers are built-in. And it does not say about whether 15.4's L2 is being used or if it's the RAW mode.

Change your commit message to say that CONFIG_IEEE802154 matches both case: either L2 or RAW mode, because it means one (or more) 15.4 device driver is being built-in.

And on your patch, remove the check on RAW_MODE